sdk/identity/azure-identity/inc/azure/identity/dll_import_export.hpp

```diff
@@ -38,6 +38,14 @@
 
 #undef AZ_IDENTITY_BUILT_AS_DLL
 
+#if defined(_azure_TESTING_BUILD)
```

Comment:
@antkmsft is there a better place to put these, within identity (and core)? This could be a temporary home for it.

Reply:
If this is temporary, the safest thing would be to have it defined in the same file that is using the macro. If there are multiple files, I'd repeat in both.

Also, the safest way for the temporary, I think, is this.
```
#if defined(_azure_TEST_BUILD)
  #if !defined(_azure_TEST_FINAL)
    #define _azure_TEST_FINAL
  #endif
  #if !defined(_azure_TEST_VIRTUAL)
    #define _azure_TEST_VIRTUAL virtual
  #endif
#else
  #if !defined(_azure_TEST_FINAL)
    #define _azure_TEST_FINAL final
  #endif
  #if !defined(_azure_TEST_VIRTUAL)
    #define _azure_TEST_VIRTUAL
  #endif
#endif
```

Because if in the GA'd version we put it in Core, and decide to change the meaning, if this specific version of Identity would be installed, it would be re-defining it unconditionally.

But that's me, I may have different view of what safe thing is.

In Core, I think it should be `<azure/core/internal/testing.hpp>`. But the person who is going to make that PR for Core is going to have the final saying on good place for it in Core anyway.

Reply:
I will keep it in this file for now but will add the `if !defined` check. We get the added benefit of doxygen ignoring this file and not documenting this, which fits our goal.